src/signaling/backend_configuration.go

```diff
@@ -148,6 +148,101 @@ func NewBackendConfiguration(config *goconf.ConfigFile) (*BackendConfiguration,
 	}, nil
 }
 
+func (b *BackendConfiguration) RemoveBackend(host string) {
+	delete(b.backends, host)
+}
+
+func (b *BackendConfiguration) UpsertHost(host string, backends []*Backend) {
+	existingIndex := 0
+	for _, existingBackend := range b.backends[host] {
+		found := false
+		index := 0
+		for _, newBackend := range backends {
+			if existingBackend == newBackend {
```

Comment:
This will compare object pointers which I think will always be different. Better would be to compare the URL of the backends.